pkg/services/govmomi/vcenter/clone_test.go

```diff
@@ -114,32 +121,268 @@ func TestGetDiskSpec(t *testing.T) {
 			cloneSpec := infrav1.VirtualMachineCloneSpec{
 				DiskGiB:            tc.cloneDiskSize,
 				AdditionalDisksGiB: tc.additionalCloneDiskSizes,
+				DataDisks:          tc.dataDisks,
 			}
 			vsphereVM := &infrav1.VSphereVM{
 				Spec: infrav1.VSphereVMSpec{
 					VirtualMachineCloneSpec: cloneSpec,
 				},
 			}
 			vmContext := &capvcontext.VMContext{VSphereVM: vsphereVM}
-			devices, err := getDiskSpec(vmContext, tc.disks)
+			deviceResults, err := getDiskSpec(vmContext, tc.disks)
 			if (tc.err != "" && err == nil) || (tc.err == "" && err != nil) || (err != nil && tc.err != err.Error()) {
 				t.Fatalf("Expected to get '%v' error from getDiskSpec, got: '%v'", tc.err, err)
 			}
-			if deviceFound := len(devices) != 0; tc.expectDevice != deviceFound {
-				t.Fatalf("Expected to get a device: %v, but got: '%#v'", tc.expectDevice, devices)
+			if deviceFound := len(deviceResults) != 0; tc.expectDevice != deviceFound {
+				t.Fatalf("Expected to get a device: %v, but got: '%#v'", tc.expectDevice, deviceResults)
 			}
 			if tc.expectDevice {
-				primaryDevice := devices[0]
+				primaryDevice := deviceResults[0]
 				validateDiskSpec(t, primaryDevice, tc.cloneDiskSize)
 				if len(tc.additionalCloneDiskSizes) != 0 {
-					secondaryDevice := devices[1]
+					secondaryDevice := deviceResults[1]
 					validateDiskSpec(t, secondaryDevice, tc.additionalCloneDiskSizes[0])
 				}
+
+				// Check number of disks present
+				if len(deviceResults) != tc.expectedDiskCount {
+					t.Fatalf("Expected device count to be %v, but found %v", tc.expectedDiskCount, len(deviceResults))
+				}
+			}
+		})
+	}
+}
+
+func TestCreateDiskSpec(t *testing.T) {
```

Comment:
```suggestion
func TestCreateDataDisks_CreateDiskSpec(t *testing.T) {
```

Reply:
Note: should we combine this and the next test as both are testing CreateDataDisks?